pkg/gui/line_by_line_panel.go

```diff
@@ -349,11 +347,7 @@ func (gui *Gui) handleOpenFileAtLine() error {
 			return errors.Errorf("unknown main context: %s", gui.State.MainContext)
 		}
 
-		// need to look at current index, then work out what my hunk's header information is, and see how far my line is away from the hunk header
-		selectedHunk := state.PatchParser.GetHunkContainingLine(state.SelectedLineIdx, 0)
-		lineNumber := selectedHunk.LineNumberOfLine(state.SelectedLineIdx)
-		filenameWithLineNum := fmt.Sprintf("%s:%d", filename, lineNumber)
-		if err := gui.OSCommand.OpenFile(filenameWithLineNum); err != nil {
+		if err := gui.OSCommand.OpenFile(filename); err != nil {
```

Comment:
the intention here was to actually open the file with the line number that was selected. We should be able to preserve that logic while defending against the panic

Reply:
@jesseduffield  Sure! Should we just keep the logic and still open the file just with `filename` then?
```go
lineNumber := selectedHunk.LineNumberOfLine(state.SelectedLineIdx)
_ = fmt.Sprintf("%s:%d", filename, lineNumber) // filename with number
if err := gui.OSCommand.OpenFile(filename); err != nil {
```

Reply:
That snippet looks like it might be the worst of both worlds: it is just as likely to panic as the existing code, but doesn't actually do anything with the line number. I think we should find out where the panic actually comes from and add some code there to defend against it

Reply:
Yeah, it makes sense. Would we defend against the panic on the `OpenFile` method then? I'm just not sure how we would do it reliably since there could exist a valid file named `a.go:123`.

Reply:
hmm, that's a tricky one. I'm okay with assuming there is no such file named `a.go:123`. Do we know why specifically the panic happens?

Reply:
The issues seems to be that we're sending the command to the OS with the line number attached to the filename.

For example, I'm looking at `line_by_line_panel.go` at line 350 and try to open the file as described in the issue, I get this:
```
2021/04/04 20:35:05 An error occurred! Please create an issue at https://github.com/jesseduffield/lazygit/issues

*errors.errorString xdg-open: file 'pkg/gui/line_by_line_panel.go:350' does not exist

github.com/jesseduffield/lazygit/pkg/commands/oscommands/os.go:205 (0x55cd4e70ab0f)
github.com/jesseduffield/lazygit/pkg/commands/oscommands/os.go:96 (0x55cd4e709cf4)
```
So the immediate fix that came to me was to not use the result of the `Sprintf`, which in this case causes the `not such file` error:
```
filenameWithLineNum := fmt.Sprintf("%s:%d", filename, lineNumber)
```
Correct me if I'm wrong but there doesn't seem to exist a standardized way to open a file at a given line number so I'm not sure how we would make this logic work.

Reply:
Interesting. In that case I would allow the user to specify an open-file-with-line-number config option which uses a go template for the line number. We could add some sensible defaults e.g. if we're going to open the program with `vi`, use whatever `vi`'s format is for specifying a file number. Otherwise just open the file without passing a number. What do you think?

Reply:
I think that would be really cool but we have some complications.
Since we're using `xdg-open`, `open` and `cmd` for the templates, passing additional arguments may not be supported.
For example:
```sh
$ xdg-open .vimrc +20
xdg-open: unexpected argument '+20'
Try 'xdg-open --help' for more information.
$ vim .vimrc +20
# opens at line 20
```
We would then need to substitute these guys (`xdg-open`...) to something like `$EDITOR` to make sure the extra arg works but I'm not sure if this change has unintentional side effects.

Another idea would be for us to add a new method, shown below, that would fix the panic but would allow us to keep the logic we had on the `line_by_line_panel.go`:
```go
func (c *OSCommand) OpenFileAtLine(filename string, lineNum int) error {
	// Intentionally ignoring the line number for now.
        commandTemplate := c.Config.GetUserConfig().OS.OpenCommand
	templateValues := map[string]string{
		"filename": c.Quote(filename),
	}

	command := utils.ResolvePlaceholderString(commandTemplate, templateValues)
	err := c.RunCommand(command)
	return err
}
```
We pass the line number as a second arg and ignore it for now.
Let me know what you think :)

Reply:
We don't need to use xdg-open, open, and cmd for the templates though. I'm saying we could add a new template entirely for when we want to open with a line number. We can have defaults set that ignore the line number, but I would always pass in the line number to the template. Does that make sense?

Reply:
Yup, it does. 
Since I won't be able to work on that for now I'll close this one so anyone can feel free to implement it :)